.github/workflows/build.yaml

```diff
@@ -1,26 +1,363 @@
+
+        - name: Download Dockerfiles
+          uses: actions/download-artifact@v3
+          with:
+            name: dockerfiles
+            path: dockerfiles
+
+        - name: Create orig file
+          run: x=(build/wake_*.tar.xz); y=${x%.tar.xz}; cp "$x" "build/${y##*/}.orig.tar.xz"
+
+        - name: Uncompress tarball
+          run: tar xvJf build/wake_*.orig.tar.xz -C build
+
+        - name: Set tests as executable
+          run: chmod -R +x build/tests
+
+        - name: Merge debian artifacts into tarball artifacts
+          run: cp -a build/debian build/wake-*
+
+        - name: Build docker image
+          run: docker build -f dockerfiles/${{ matrix.dockerfile }} ${{ matrix.extra_docker_build_args }} -t wake-${{ matrix.dockerfile }} .
+
+        - name: Build
+          run: docker run --rm --mount type=bind,source=$PWD/build,target=/build --device /dev/fuse --cap-add SYS_ADMIN --security-opt apparmor:unconfined ${{ matrix.extra_docker_run_args }} wake-${{ matrix.dockerfile }} /bin/sh -c '${{ matrix.build_cmd }}'
+
+        - name: Test
+          run: docker run --rm --mount type=bind,source=$PWD/build,target=/build --device /dev/fuse --cap-add SYS_ADMIN --security-opt apparmor:unconfined wake-${{ matrix.dockerfile }} /bin/sh -c '${{ matrix.test_cmd }}'
+
+        - name: Install
+          run: install -D -t release/${{ matrix.target }} ${{ matrix.install_src_glob }}
+
+        - name: Upload
+          uses: actions/upload-artifact@v3
+          with:
+            name: release_${{ matrix.target }}
+            path: release/${{ matrix.target }}
+
+    macos_10_15:
+        name: MacOS 10.15 Build
         runs-on: macos-10.15
 
         steps:
-            - name: "Clone wake"
+            - name: Clone wake
               uses: actions/checkout@v2
               with:
                 fetch-depth: 0
 
-            - name: "Install deps"
+            - name: Install deps
               run: |
                 brew install --cask osxfuse
                 brew install re2 || brew install re2 --head
 
-            - name: "Build wake"
+            - name: Build wake
               run: make all -j3
 
-            - name: "Test wake"
+            - name: Test wake
               run: make test
+
+    docs:
+      runs-on: ubuntu-latest
+      needs: tarball
+      name: Docs
+      steps:
+        - name: Download index.html
+          uses: actions/download-artifact@v3
+          with:
+            name: index.html
+
+        - name: Download html
+          uses: actions/download-artifact@v3
+          with:
+            name: html
+
+        - name: Make .nojekyll
+          run: touch .nojekyll
+
+        - name: Upload
+          uses: actions/upload-pages-artifact@v1
+          with:
+            path: .
+
+    deploy_docs:
+      # Only run the 'deploy_docs' job if this workflow was triggered by a push to master
+      if: github.event_name == 'push' && github.ref_type == 'branch' && github.ref_name == 'master'
```

Comment:
I assume this catches PRs being merged as well?

Reply:
Right so the whole workflow triggers on any of 

1) PR opened
2) commit pushed to master
3) Tag starting with a `v` created and pushed

The filter for this specific job says only run when a commit is pushed to master, and since we only let commits land on master via a PR this runs right after a PR is merged